agent/acs/handler/acs_handler.go

```diff
@@ -189,70 +189,59 @@ func NewSession(
 // Start starts the session. It'll forever keep trying to connect to ACS unless
 // the context is cancelled.
 //
-// If the context is cancelled, Start() would return with the error code returned
-// by the context.
+// If the context is cancelled, Start() would return nil
```

Comment:
seems like it returns nil unconditionally. Might as well remove the return type.